libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2998,15 +2995,22 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 						converted := <arrayStorageToMem>(value)
 					</fromStorage>
 					<?fromCalldata>
-						converted := <allocateMemoryArray>(length)
-						<copyToMemory>(value, add(converted, 0x20), length)
+						converted := <allocateMemoryArray>(<length>)
+						<copyToMemory>(
```

Comment:
I don't think this will work. What about an array of structs? Shouldn't this just call the appropriate `abiDecode` function?

Reply:
I’ll check test coverage and add tests for nested arrays and all combinations with structs, as we did with copying from calldata/memory to storage

And then check for each specific case